blocks/tags-content-source-block/sources/tags-api.js

```diff
@@ -1,8 +1,26 @@
+const make404 = () => {
+  const error = new Error('Not found');
+  error.statusCode = 404;
+  throw error;
+};
+
 export default {
   resolve({ slug }) {
     return `/tags/v2/slugs?slugs=${slug}`;
   },
   params: {
     slug: 'text',
   },
+  transform: (data) => {
+    if (!data || !data.Payload) {
+      make404();
```

Comment:
would you want to return here as well?

Reply:
```suggestion
      return make404();
```

Reply:
the return at the end is only to comply with eslint, is not needed because the function never returns, throw an error

Reply:
right right